src/app/clusters/thread-border-router-management-server/thread-border-router-management-server.cpp

```
@@ -0,0 +1,384 @@
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::ReadBorderAgentID(MutableByteSpan & borderAgentId)
+{
+    ReturnErrorOnFailure(mDelegate->GetBorderAgentId(borderAgentId));
+    ReturnErrorCodeIf(borderAgentId.size() != kBorderAgentIdLength, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::ReadThreadVersion(uint16_t & threadVersion)
+{
+    return mDelegate->GetThreadVersion(threadVersion);
+}
+
+CHIP_ERROR ServerInstance::ReadInterfaceEnabled(bool & interfaceEnabled)
+{
+    return mDelegate->GetInterfaceEnabled(interfaceEnabled);
+}
+
+CHIP_ERROR ServerInstance::ReadActiveDatasetTimestamp(Optional<uint64_t> & activeDatasetTimestamp)
+{
+    uint64_t activeDatasetTimestampValue;
+    Thread::OperationalDataset activeDataset;
+    if (mDelegate->GetDataset(activeDataset, Delegate::DatasetType::kActive) == CHIP_NO_ERROR &&
+        activeDataset.GetActiveTimestamp(activeDatasetTimestampValue) == CHIP_NO_ERROR)
+    {
+        activeDatasetTimestamp.SetValue(activeDatasetTimestampValue);
+    }
+    else
+    {
+        activeDatasetTimestamp.ClearValue();
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    if (aPath.mClusterId != ThreadBorderRouterManagement::Id)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    if (!mDelegate)
+    {
+        ChipLogError(Zcl, "Thread Border Router Management server not initialized");
+        return CHIP_ERROR_UNINITIALIZED;
+    }
+    CHIP_ERROR status = CHIP_NO_ERROR;
+    switch (aPath.mAttributeId)
+    {
+    case Globals::Attributes::FeatureMap::Id: {
+        BitFlags<Feature> featureMap;
+        status = ReadFeatureMap(featureMap);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(featureMap);
+        }
```

All the checks for `if (status == CHIP_NO_ERROR)` that lead to not touching the encoder mean that if there's anything wrong internally it will all silently fail. Please refactor to remove a lot of the CHIP_ERROR paths and if any are left, make sure that an error status of `Failure` is encoded in every corner where CHIP_ERROR is not success and it was not expected.

---

If there is anything wrong internally this function will return the failure status so that the reporter will encode a failure status in the report for the read action.

---

There is no good guarantee today of which CHIP_ERROR will be encoded.

---

#34607